services/userlog/pkg/service/service.go

```diff
@@ -378,49 +315,99 @@ func (ul *UserlogService) gatherSpaceMembers(ctx context.Context, space *storage
 	return users, nil
 }
 
+func (ul *UserlogService) resolveID(ctx context.Context, userid *user.UserId, groupid *group.GroupId) ([]string, error) {
+	if userid != nil {
+		return []string{userid.GetOpaqueId()}, nil
+	}
+
+	return ul.resolveGroup(ctx, groupid.GetOpaqueId())
+}
+
 // resolves the users of a group
 func (ul *UserlogService) resolveGroup(ctx context.Context, groupID string) ([]string, error) {
-	if ctx == nil {
-		return nil, errors.New("need authenticated context to resolve groups")
+	grp, err := ul.getGroup(ctx, groupID)
+	if err != nil {
+		return nil, err
 	}
 
-	r, err := ul.gwClient.GetGroup(ctx, &group.GetGroupRequest{GroupId: &group.GroupId{OpaqueId: groupID}})
+	var userIDs []string
+	for _, m := range grp.GetMembers() {
+		userIDs = append(userIDs, m.GetOpaqueId())
+	}
+
+	return userIDs, nil
+}
+
+func (ul *UserlogService) impersonate(u *user.UserId) context.Context {
+	if u == nil {
+		ul.log.Debug().Msg("cannot impersonate nil user")
+		return nil
+	}
+
+	ctx, _, err := utils.Impersonate(u, ul.gwClient, ul.cfg.MachineAuthAPIKey)
+	if err != nil {
+		ul.log.Error().Err(err).Str("userid", u.GetOpaqueId()).Msg("failed to impersonate user")
+		return nil
+	}
+	return ctx
+}
+
+func (ul *UserlogService) getSpace(ctx context.Context, spaceID string) (*storageprovider.StorageSpace, error) {
+	res, err := ul.gwClient.ListStorageSpaces(ctx, listStorageSpaceRequest(spaceID))
 	if err != nil {
 		return nil, err
 	}
 
-	if r.GetStatus().GetCode() != rpc.Code_CODE_OK {
-		return nil, fmt.Errorf("unexpected status code from gateway client: %d", r.GetStatus().GetCode())
+	if res.GetStatus().GetCode() != rpc.Code_CODE_OK {
+		return nil, fmt.Errorf("Unexpected status code while getting space: %v", res.GetStatus().GetCode())
```

Comment:
Maybe build the error from code and message?
```suggestion
		return nil, fmt.Errorf("Error while getting space: (%v) %s", res.GetStatus().GetCode(), res.GetStatus().GetMessage())
```

Reply:
OK. I'll fix it here: https://github.com/owncloud/ocis/pull/5753